crocket/src/main/java/com/crocket/LevelView.java

```diff
@@ -1,70 +1,142 @@
 package com.crocket;
 
 import java.awt.*;
-import java.awt.image.*;
-import java.io.File;
+import java.awt.image.BufferedImage;
 import java.io.IOException;
-import java.util.HashMap;
-import java.util.Map;
 import java.util.Set;
-
-import javax.imageio.ImageIO;
+import javax.swing.ImageIcon;
+import javax.swing.JLabel;
+import javax.swing.JLayeredPane;
 import javax.swing.JOptionPane;
-import javax.swing.JPanel;
 
-public class LevelView extends JPanel implements ILevelView {
+public class LevelView extends JLayeredPane implements ILevelView {
     private Surface[][] levelSurfacemap;
-    private Map<String, BufferedImage> textureCacheMap = new HashMap<>();
+    private TextureManager textureManager;
     private int levelHeight;
     private int levelWidth;
 
-
+    public LevelView() {
+        setDoubleBuffered(true);
+        this.textureManager = new TextureManager();
+        this.textureManager.loadTextures();
+    }
 
     @Override
     protected void paintComponent(Graphics g) {
+        super.paintComponent(g);
         try {
             drawSurfaces(g);
         } catch (IOException | NullPointerException e) {
-            JOptionPane.showMessageDialog(null, e.getMessage(), "Error", JOptionPane.ERROR_MESSAGE);
+            JOptionPane.showMessageDialog(null, e.toString(), "Error", JOptionPane.ERROR_MESSAGE);
             System.exit(1);
         }
     }
 
-    private void drawSurfaces(Graphics g) throws IOException {
+    private void drawSurfaceTile(Graphics g, Surface surface, int x, int y) throws IOException {
+        g.drawImage(textureManager.getTexture(surface.name()), x * 100, y * 100, this);
+    }
+
+    public void drawSurfaces(Graphics g) throws IOException {
         for (int y = 0; y < levelHeight; y++) {
             for (int x = 0; x < levelWidth; x++) {
-                getTileTexture(g, y, x);
+                Surface surface = levelSurfacemap[y][x];
+                drawSurfaceTile(g, surface, x, y);
             }
         }
     }
+
+    private void drawLowerLayerEntity(DrawableEntity entity, BufferedImage bufferedImage) {
+        JLabel label = makeLabel(entity, bufferedImage);
+        add(label, JLayeredPane.DEFAULT_LAYER);
+    }
+
+    private void drawMiddleLayerEntity(DrawableEntity entity, BufferedImage bufferedImage) {
+        JLabel label = makeLabel(entity, bufferedImage);
+        add(label, JLayeredPane.MODAL_LAYER);
+    }
```

Comment:
What is the purpose of dividing the methods in Middle and Lower, what's the different? Just a question

Reply:
It is just the add method that is different.  I wanted to use a JLayeredPane so that we can draw entities on layers.